File: haskell/raindrops/src/Raindrops.hs
module Raindrops (convert) where

convert :: Int -> String
convert x =
    let
        -- If the number has 3 as a factor, output 'Pling'.
        -- If the number has 5 as a factor, output 'Plang'.
        -- If the number has 7 as a factor, output 'Plong'.

        noise factor pliaong =
            if (x `mod` factor) == 0 then
                pliaong
            else
                ""            

        noises =
            (noise 3 "Pling") ++
            (noise 5 "Plang") ++
            (noise 7 "Plong")
    in
        if not $ null noises then
            noises
        else
            show x

File: go/raindrops/raindrops.go
package raindrops

import "fmt"

const testVersion = 2

// Convert number to raindrop sound
func Convert(x int) string {
	var pling, plang, plong, sound string
	if (x % 3) == 0 {
		pling = "Pling"
	}
	if (x % 5) == 0 {
		plang = "Plang"
	}
	if (x % 7) == 0 {
		plong = "Plong"
	}
	sound = pling + plang + plong
	if len(sound) > 0 {
		return sound
	}
	return fmt.Sprintf("%d", x)
}
